account_analytic_project/model/account_analytic_account.py

```diff
@@ -0,0 +1,78 @@
+# -*- encoding: utf-8 -*-
+"""Extend account.analytic.account."""
+##############################################################################
+#
+#    OpenERP, Open Source Management Solution
+#    This module copyright (C) 2015 Therp BV <http://therp.nl>
+#    All Rights Reserved
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+from openerp import _, api, models, fields
+
+
+class AccountAnalyticAccount(models.Model):
+    _inherit = 'account.analytic.account'
+
+    @api.multi
+    def _compute_has_projects(self):
+        """Check wether any project for this analytic account."""
+        for this in self:
+            this.has_projects = bool(this.project_ids)
+
+    @api.multi
+    def create_project_for_account(self):
+        """Create a project for an already existing analytic account."""
+        project_model = self.env['project.project']
+        for this in self:
+            if not this.has_projects and this.type != 'view':
+                project_model.create({
+                    'name': this.name,
+                    'analytic_account_id': this.id,
+                })
+        return True
```

Comment:
To provide the code for the button to create a project? Not really intuitive to check "use tasks" in order to indirectly create a project. And you might want a project, but no tasks.